docs/developer-guide/implement-a-udf.rst

```diff
@@ -0,0 +1,349 @@
+.. _implement-a-udf:
+
+Implement a User-defined Function (UDF)
+#######################################
+
+Prerequisites
+     - `Apache Maven <https://maven.apache.org/download.cgi>`__
+     - |cp| installed locally
+     - Internet connectivity for downloading Confluent POM files
+
+Create the KSQL Extensions Directory
+************************************
+
+When you create a custom user-defined function (UDF), you implement it in Java
+and deploy it as a JAR to the KSQL extensions directory. By default, this 
+directory doesn't exist, so you need to create it and assign it in the KSQL
+Server configuration properties.
+
+Create the KSQL extensions directory, ``<path-to-confluent>/etc/ksql/ext``:
+
+.. codewithvars:: bash
+
+    mkdir confluent-|release|/etc/ksql/ext
+
+Edit the ``ksql-server.properties`` configuration file in
+``<path-to-confluent>/etc/ksql`` to add the fully qualified path to the
+``ext`` directory:
+
+.. codewithvars:: text
+
+    ksql.extension.dir=/home/my-home-dir/confluent-|release|/etc/ksql/ext
+
+.. note::
+
+    Use the fully qualified path. KSQL Server won't load extensions if the 

```

Comment:
It _should_ work with a path relative to the server installation directory, e.g. something like the following might work:

`../ext`


If it doesn't, we should make it work.

Reply:
Relative path does work! Path with `~` doesn't seem to work.